cpp/core/operators/shuffle/splitter.cc

```diff
@@ -1150,7 +1150,7 @@ arrow::Status Splitter::SplitBinaryType(
     auto size = reducer_offset_offset_[pid + 1] - r;
 
     auto multiply = 1;
-    for (register uint32_t x = 0; x < size; x++) {
+    for (uint32_t x = 0; x < size; x++) {
```

Comment:
Why do we need this change?

Reply:
Because the register storage class is deprecated from c++17, I modified the corresponding code using register, it does not matter.

--
I explained in the above